### ISSUE TITLE
unexpected token: '<EOF>' (Line: 1) on class creation

### ISSUE THREAD

**Issue by dbleier** [joeferraro/MavensMate-SublimeText#567]
Recently whenever I try to create a new Apex Class via MavensMate > Metadata > New Apex Class (or trigger), I get the error
xxxx.cls: unexpected token: '<EOF>' (Line: 1)

It creates a blank file but without the normal scaffolding. If I put in the complete code a save, it works but it's annoying. This only started in the last few days.

Using MM 4.0.4
Sublime Text Build 3083 on Ubuntu 14.04

Aside from that MM works wonderfully, thanks so much.

Dovid

**Comment by ralphcallaway** [comment]
able to reproduce on Ubuntu 14.04

mm-new_metadata.log
```
--------------------------------------------
---------- NEW OPERATION REQUESTED ---------
--------------------------------------------

new_metadata
Namespace(client='SUBLIME_TEXT_3', format='json', offline=False, operation='new_metadata', quiet=False, respond_with_html=False, ui_switch=False, uid=None, verbose=False)
{u'github_template': {u'file_name': u'ApexClass.cls', u'author': u'MavensMate', u'params': [{u'default': u'MyApexClass', u'description': u'Apex Class API Name', u'name': u'api_name'}], u'description': u'The default template for an Apex Class', u'name': u'Default'}, u'project_name': u'Dev', 'verbose': False, 'args': Namespace(client='SUBLIME_TEXT_3', format='json', offline=False, operation='new_metadata', quiet=False, respond_with_html=False, ui_switch=False, uid=None, verbose=False), 'client': 'SUBLIME_TEXT_3', u'params': {u'api_name': u'MyApexClass1'}, u'workspace': u'/media/psf/Home/MavensMate/MMUbuntuWorkspace', u'metadata_type': u'ApexClass', 'ui': False, 'operation': 'new_metadata'}

--------------------------------------------
creds!
<redacted>
production
endpont is ::: 
https://www.salesforce.com/services/Soap/u/31.0
GETTING USER INFO
<redcated>
LOGIN RESULT
<redacted>
storing updated session information locally
get_package_types
project
project
retrieve request: 
{u'ApexClass': [u'MyApexClass1']}
----> 
{'name': u'ApexClass', 'members': [u'MyApexClass1']}
{u'inFolder': False, u'xmlName': u'ApexClass', u'suffix': u'cls', u'directoryName': u'classes', u'metaFile': True}
---request payload---
{'apiVersion': '31.0', 'unpackaged': {'types': [{'name': u'ApexClass', 'members': [u'MyApexClass1']}]}}
result of retrieve: 


(AsyncResult){
   done = False
   id = "09S400000002t8fEAA"
   state = "Queued"
 }
---> result is not done
waiting for retrieve request ...
checkStatusResponse --->
(RetrieveResult){
   done = True
   fileProperties[] = 
      (FileProperties){
         createdById = "00540000000mbcuAAA"
         createdByName = "Ralph Callaway"
         createdDate = 2015-04-25 14:36:59.000704
         fileName = "unpackaged/package.xml"
         fullName = "unpackaged/package.xml"
         id = None
         lastModifiedById = "00540000000mbcuAAA"
         lastModifiedByName = "Ralph Callaway"
         lastModifiedDate = 2015-04-25 14:36:59.000704
         manageableState = "unmanaged"
         type = "Package"
      },
   id = "09S400000002t8fEAA"
   messages[] = 
      (RetrieveMessage){
         fileName = "unpackaged/package.xml"
         problem = "Entity of type 'ApexClass' named 'MyApexClass1' cannot be found"
      },
   status = "Succeeded"
   success = True
   zipFile = "UEsDBBQACAgIAJ2smUYAAAAAAAAAAAAAAAAWAAAAdW5wYWNrYWdlZC9wYWNrYWdlLnhtbE2OywrCMBBF9/mKkL2dVEVE0hQR3Aku9ANiHGuxedAEaf/e0Ic6qznD4c4VZWca+sY21M4WLM84o2i1u9e2Ktj1clxsWSmJOCv9UhXSZNtQsGeMfgcQnPJZeLhWY6adgSXnG+BrMBjVXUXFJKFpROw9hnEf2KC5pZfy1O89dodGhZALmK8/zyqD8qsIGHiMhL9MMfWXq9RfwExEwFRbkg9QSwcIwnB+saAAAADoAAAAUEsBAhQAFAAICAgAnayZRsJwfrGgAAAA6AAAABYAAAAAAAAAAAAAAAAAAAAAAHVucGFja2FnZWQvcGFja2FnZS54bWxQSwUGAAAAAAEAAQBEAAAA5AAAAAAA"
 }
---------------------
result from command execution
{"checkOnly": false, "completedDate": "2015-04-25T21:36:57.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-04-25T21:36:54.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-04-25T21:36:56.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass1.cls", "fullName": "MyApexClass1", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-04-25T21:36:57.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000utaLCAQ", "ignoreWarnings": false, "lastModifiedDate": "2015-04-25T21:36:57.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-04-25T21:36:55.000Z", "status": "Failed", "success": false}
<type 'str'>
---------------------


---------------------
RESPONDING TO REQUEST
---------------------


{"checkOnly": false, "completedDate": "2015-04-25T21:36:57.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-04-25T21:36:54.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-04-25T21:36:56.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass1.cls", "fullName": "MyApexClass1", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-04-25T21:36:57.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000utaLCAQ", "ignoreWarnings": false, "lastModifiedDate": "2015-04-25T21:36:57.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-04-25T21:36:55.000Z", "status": "Failed", "success": false}
```

mm console
```
Operation: Creating New ApexClass: MyApexClass1
Timestamp: Sat, 25 Apr 2015 15:36:48
   Result: [OPERATION FAILED]: Whoops, unable to parse the response. Please enable logging (http://mavensmate.com/Plugins/Sublime_Text/Plugin_Logging) and post relevant log(s) to a new issue at https://github.com/joeferraro/MavensMate-SublimeText
[RESPONSE FROM MAVENSMATE]: /usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion 'g_type_from_name (name) == 0' failed

  import gobject._gobject

ERROR:suds.client:<?xml version="1.0" encoding="UTF-8"?>

<SOAP-ENV:Envelope xmlns:ns0="urn:partner.soap.sforce.com" xmlns:ns1="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:tns="urn:partner.soap.sforce.com" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">

   <SOAP-ENV:Header>

      <tns:SessionHeader>

         <tns:sessionId>redacted</tns:sessionId>

      </tns:SessionHeader>

   </SOAP-ENV:Header>

   <ns1:Body>

      <ns0:getUserInfo/>

   </ns1:Body>

</SOAP-ENV:Envelope>

{"checkOnly": false, "completedDate": "2015-04-25T21:36:57.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-04-25T21:36:54.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-04-25T21:36:56.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass1.cls", "fullName": "MyApexClass1", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-04-25T21:36:57.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000utaLCAQ", "ignoreWarnings": false, "lastModifiedDate": "2015-04-25T21:36:57.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-04-25T21:36:55.000Z", "status": "Failed", "success": false}
 FAILED]: /usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion 'g_type_from_name (name) == 0' failed

  import gobject._gobject

ERROR:suds.client:<?xml version="1.0" encoding="UTF-8"?>

<SOAP-ENV:Envelope xmlns:ns0="urn:partner.soap.sforce.com" xmlns:ns1="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:tns="urn:partner.soap.sforce.com" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">

   <SOAP-ENV:Header>

      <tns:SessionHeader>

         <tns:sessionId>redacted</tns:sessionId>

      </tns:SessionHeader>

   </SOAP-ENV:Header>

   <ns1:Body>

      <ns0:getUserInfo/>

   </ns1:Body>

</SOAP-ENV:Envelope>

{"checkOnly": false, "completedDate": "2015-04-25T21:36:57.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-04-25T21:36:54.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-04-25T21:36:56.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass1.cls", "fullName": "MyApexClass1", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-04-25T21:36:57.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000utaLCAQ", "ignoreWarnings": false, "lastModifiedDate": "2015-04-25T21:36:57.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-04-25T21:36:55.000Z", "status": "Failed", "success": false}
```

st 3 console
```
[MAVENSMATE]: [{'description': 'Apex Class API Name', 'name': 'api_name', 'default': 'MyApexClass'}]
[MAVENSMATE]: [{'description': 'Apex Class API Name', 'name': 'api_name', 'default': 'MyApexClass'}]
[MAVENSMATE]: Calling mm_interface
[MAVENSMATE]: OPERATION: new_metadata
[MAVENSMATE]: {'params': {'metadata_type': 'ApexClass', 'github_template': {'description': 'The default template for an Apex Class', 'name': 'Default', 'author': 'MavensMate', 'file_name': 'ApexClass.cls', 'params': [{'description': 'Apex Class API Name', 'name': 'api_name', 'default': 'MyApexClass'}]}, 'params': {'api_name': 'MyApexClass1'}}}
[MAVENSMATE]: mm.py command: 
[MAVENSMATE]: "/usr/bin/python" "/media/psf/Home/MavensMate/mm/mm.py" -c SUBLIME_TEXT_3 -o new_metadata
[MAVENSMATE]: {"metadata_type": "ApexClass", "workspace": "/media/psf/Home/MavensMate/MMUbuntuWorkspace", "github_template": {"description": "The default template for an Apex Class", "name": "Default", "author": "MavensMate", "file_name": "ApexClass.cls", "params": [{"description": "Apex Class API Name", "name": "api_name", "default": "MyApexClass"}]}, "params": {"api_name": "MyApexClass1"}, "project_name": "Dev"}
[MAVENSMATE]: response from mm: /usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion 'g_type_from_name (name) == 0' failed

  import gobject._gobject

ERROR:suds.client:<?xml version="1.0" encoding="UTF-8"?>

<SOAP-ENV:Envelope xmlns:ns0="urn:partner.soap.sforce.com" xmlns:ns1="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:tns="urn:partner.soap.sforce.com" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">

   <SOAP-ENV:Header>

      <tns:SessionHeader>

         <tns:sessionId>redacted</tns:sessionId>

      </tns:SessionHeader>

   </SOAP-ENV:Header>

   <ns1:Body>

      <ns0:getUserInfo/>

   </ns1:Body>

</SOAP-ENV:Envelope>

{"checkOnly": false, "completedDate": "2015-04-25T21:36:57.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-04-25T21:36:54.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-04-25T21:36:56.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass1.cls", "fullName": "MyApexClass1", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-04-25T21:36:57.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000utaLCAQ", "ignoreWarnings": false, "lastModifiedDate": "2015-04-25T21:36:57.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-04-25T21:36:55.000Z", "status": "Failed", "success": false}

[MAVENSMATE]: HANDLING COMPILE!
[MAVENSMATE]: /usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion 'g_type_from_name (name) == 0' failed

  import gobject._gobject

ERROR:suds.client:<?xml version="1.0" encoding="UTF-8"?>

<SOAP-ENV:Envelope xmlns:ns0="urn:partner.soap.sforce.com" xmlns:ns1="http://schemas.xmlsoap.org/soap/envelope/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:tns="urn:partner.soap.sforce.com" xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/">

   <SOAP-ENV:Header>

      <tns:SessionHeader>

         <tns:sessionId>redacted</tns:sessionId>

      </tns:SessionHeader>

   </SOAP-ENV:Header>

   <ns1:Body>

      <ns0:getUserInfo/>

   </ns1:Body>

</SOAP-ENV:Envelope>

{"checkOnly": false, "completedDate": "2015-04-25T21:36:57.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-04-25T21:36:54.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-04-25T21:36:56.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass1.cls", "fullName": "MyApexClass1", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-04-25T21:36:57.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000utaLCAQ", "ignoreWarnings": false, "lastModifiedDate": "2015-04-25T21:36:57.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-04-25T21:36:55.000Z", "status": "Failed", "success": false}

[MAVENSMATE]: string indices must be integers
Traceback (most recent call last):
  File "/home/parallels/.config/sublime-text-3/Packages/MavensMate/lib/mm_response_handlers.py", line 184, in __handle_compile_response
    elif 'success' in self.result and util.to_bool(self.result['success']) == False and (('messages' in self.result or 'Messages' in self.result) or 'details' in self.result):
TypeError: string indices must be integers
[MAVENSMATE]: None
[MAVENSMATE]: <class 'str'>
```

**Comment by ralphcallaway** [comment]
also happening on windows 7, not an issue on mac os x

guessing this is an issue with the updated template support pull request, may need to revert

@dbleier short term if you want to uninstall the mavens mate package and install manually if you go to v4.0.3 you should stop having this issue

**Comment by ralphcallaway** [comment]
confirmed issue not present in v4.0.3

**Comment by ralphcallaway** [comment]
looks like it was introduced by pull #427

**Comment by ralphcallaway** [comment]
looks like we're missing a dependent update to mm core, see https://github.com/joeferraro/mm/pull/30

going to need to rollback private repos until that is also merged

**Comment by dbleier** [comment]
@ralphcallaway, I followed your suggestion and manually switched to 4.0.3 and all is working. Only thing is it's labelled as 4.0.2 in packages.json, but maybe they just forgot to update the version number.

thanks a lot and looking forward to the new updated fix.

**Comment by ralphcallaway** [comment]
@dbleier good to know, and yeah, the version number in package.json is incorrect for v4.0.3

**Comment by ralphcallaway** [comment]
fix in place, but issue is still occurring on linux platforms and appears to have already been present in v4.0.3, may be related to #572 gumming up the works

**Comment by ralphcallaway** [comment]
linux debugging details

mm console
```
Operation: Creating New ApexClass: MyApexClass3
Timestamp: Mon, 04 May 2015 08:28:38
   Result: [OPERATION FAILED]: Whoops, unable to parse the response. Please enable logging (http://mavensmate.com/Plugins/Sublime_Text/Plugin_Logging) and post relevant log(s) to a new issue at https://github.com/joeferraro/MavensMate-SublimeText
[RESPONSE FROM MAVENSMATE]: /usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion 'g_type_from_name (name) == 0' failed

  import gobject._gobject

{"checkOnly": false, "completedDate": "2015-05-04T14:28:43.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-05-04T14:28:42.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-05-04T14:28:43.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass3.cls", "fullName": "MyApexClass3", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-05-04T14:28:43.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000vMBjCAM", "ignoreWarnings": false, "lastModifiedDate": "2015-05-04T14:28:43.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-05-04T14:28:43.000Z", "status": "Failed", "success": false}
 FAILED]: /usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion 'g_type_from_name (name) == 0' failed

  import gobject._gobject

{"checkOnly": false, "completedDate": "2015-05-04T14:28:43.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-05-04T14:28:42.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-05-04T14:28:43.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass3.cls", "fullName": "MyApexClass3", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-05-04T14:28:43.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000vMBjCAM", "ignoreWarnings": false, "lastModifiedDate": "2015-05-04T14:28:43.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-05-04T14:28:43.000Z", "status": "Failed", "success": false}
```

st3 console
```
[MAVENSMATE]: HANDLING COMPILE!
[MAVENSMATE]: /usr/lib/python2.7/dist-packages/gobject/constants.py:24: Warning: g_boxed_type_register_static: assertion 'g_type_from_name (name) == 0' failed

  import gobject._gobject

{"checkOnly": false, "completedDate": "2015-05-04T14:28:43.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-05-04T14:28:42.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-05-04T14:28:43.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass3.cls", "fullName": "MyApexClass3", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-05-04T14:28:43.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000vMBjCAM", "ignoreWarnings": false, "lastModifiedDate": "2015-05-04T14:28:43.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-05-04T14:28:43.000Z", "status": "Failed", "success": false}

[MAVENSMATE]: string indices must be integers
Traceback (most recent call last):
  File "/home/parallels/.config/sublime-text-3/Packages/MavensMate/lib/mm_response_handlers.py", line 184, in __handle_compile_response
    elif 'success' in self.result and util.to_bool(self.result['success']) == False and (('messages' in self.result or 'Messages' in self.result) or 'details' in self.result):
TypeError: string indices must be integers
[MAVENSMATE]: None
[MAVENSMATE]: <class 'str'>
```

new metadata log
```
--------------------------------------------
---------- NEW OPERATION REQUESTED ---------
--------------------------------------------

new_metadata
Namespace(client='SUBLIME_TEXT_3', format='json', offline=False, operation='new_metadata', quiet=False, respond_with_html=False, ui_switch=False, uid=None, verbose=False)
{u'github_template': {u'file_name': u'ApexClass.cls', u'author': u'MavensMate', u'params': [{u'default': u'MyApexClass', u'description': u'Apex Class API Name', u'name': u'api_name'}], u'description': u'The default template for an Apex Class', u'name': u'Default'}, u'project_name': u'Dev', 'verbose': False, 'args': Namespace(client='SUBLIME_TEXT_3', format='json', offline=False, operation='new_metadata', quiet=False, respond_with_html=False, ui_switch=False, uid=None, verbose=False), 'client': 'SUBLIME_TEXT_3', u'params': {u'api_name': u'MyApexClass3'}, u'workspace': u'/media/psf/Home/MavensMate/MMUbuntuWorkspace', u'metadata_type': u'ApexClass', 'ui': False, 'operation': 'new_metadata'}

--------------------------------------------
creds! <redacted>
production
endpont is ::: 
https://www.salesforce.com/services/Soap/u/31.0
GETTING USER INFO
<redacted>
get_package_types
project
project
retrieve request: 
{u'ApexClass': [u'MyApexClass3']}
----> 
{'name': u'ApexClass', 'members': [u'MyApexClass3']}
{u'inFolder': False, u'xmlName': u'ApexClass', u'suffix': u'cls', u'directoryName': u'classes', u'metaFile': True}
---request payload---
{'apiVersion': '31.0', 'unpackaged': {'types': [{'name': u'ApexClass', 'members': [u'MyApexClass3']}]}}
result of retrieve: 


(AsyncResult){
   done = False
   id = "09S400000002tayEAA"
   state = "Queued"
 }
---> result is not done
waiting for retrieve request ...
checkStatusResponse --->
(RetrieveResult){
   done = True
   fileProperties[] = 
      (FileProperties){
         createdById = "00540000000mbcuAAA"
         createdByName = "Ralph Callaway"
         createdDate = 2015-05-04 07:28:45.000643
         fileName = "unpackaged/package.xml"
         fullName = "unpackaged/package.xml"
         id = None
         lastModifiedById = "00540000000mbcuAAA"
         lastModifiedByName = "Ralph Callaway"
         lastModifiedDate = 2015-05-04 07:28:45.000643
         manageableState = "unmanaged"
         type = "Package"
      },
   id = "09S400000002tayEAA"
   messages[] = 
      (RetrieveMessage){
         fileName = "unpackaged/package.xml"
         problem = "Entity of type 'ApexClass' named 'MyApexClass3' cannot be found"
      },
   status = "Succeeded"
   success = True
   zipFile = "UEsDBBQACAgIAJZzpEYAAAAAAAAAAAAAAAAWAAAAdW5wYWNrYWdlZC9wYWNrYWdlLnhtbE2OywrCMBBF9/mKkL2daEVE0hQR3Aku9ANiHGuxedAEaf/e0Ic6qznD4c4VZWca+sY21M4WbJlxRtFqd69tVbDr5bjYslIScVb6pSqkybahYM8Y/Q4gOOWz8HCtxkw7AyvON8DXYDCqu4qKSULTiNh7DOM+sEFzSy/lqd977A6NCiEXMF9/nlUG5VcRMPAYCX+ZYuov89RfwExEwFRbkg9QSwcImnO6faAAAADoAAAAUEsBAhQAFAAICAgAlnOkRppzun2gAAAA6AAAABYAAAAAAAAAAAAAAAAAAAAAAHVucGFja2FnZWQvcGFja2FnZS54bWxQSwUGAAAAAAEAAQBEAAAA5AAAAAAA"
 }
---------------------
result from command execution
{"checkOnly": false, "completedDate": "2015-05-04T14:28:43.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-05-04T14:28:42.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-05-04T14:28:43.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass3.cls", "fullName": "MyApexClass3", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-05-04T14:28:43.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000vMBjCAM", "ignoreWarnings": false, "lastModifiedDate": "2015-05-04T14:28:43.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-05-04T14:28:43.000Z", "status": "Failed", "success": false}
<type 'str'>
---------------------


---------------------
RESPONDING TO REQUEST
---------------------


{"checkOnly": false, "completedDate": "2015-05-04T14:28:43.000Z", "createdBy": "00540000000mbcu", "createdByName": "Ralph Callaway", "createdDate": "2015-05-04T14:28:42.000Z", "details": {"componentFailures": {"changed": false, "componentType": "ApexClass", "created": false, "createdDate": "2015-05-04T14:28:43.000Z", "deleted": false, "fileName": "unpackaged/classes/MyApexClass3.cls", "fullName": "MyApexClass3", "lineNumber": "1", "problem": "unexpected token: '<EOF>'", "problemType": "Error", "success": false}, "componentSuccesses": {"changed": true, "componentType": null, "created": false, "createdDate": "2015-05-04T14:28:43.000Z", "deleted": false, "fileName": "unpackaged/package.xml", "fullName": "package.xml", "success": true}, "runTestResult": {"numFailures": "0", "numTestsRun": "0", "totalTime": "0.0"}}, "done": true, "id": "0Af40000000vMBjCAM", "ignoreWarnings": false, "lastModifiedDate": "2015-05-04T14:28:43.000Z", "numberComponentErrors": "1", "numberComponentsDeployed": "0", "numberComponentsTotal": "1", "numberTestErrors": "0", "numberTestsCompleted": "0", "numberTestsTotal": "0", "rollbackOnError": true, "runTestsEnabled": false, "startDate": "2015-05-04T14:28:43.000Z", "status": "Failed", "success": false}
```

**Comment by ralphcallaway** [comment]
@dbleier fix should be in place for this in v4.0.5-prerelease

can you turn on pre-releases for mavensmate and let me know if that fixes you're issue? see install_prereleases setting for package control in https://packagecontrol.io/docs/settings

please note there can be a delay of an hour or two before new releases are available in the package control repos (and longer if you have adjusted your cache settings locally)

**Comment by dbleier** [comment]
@ralphcallaway is v4.0.5-prerelease otherwise stable? I don't want to install and then have all sorts of other issues. Since I rolled back to v4.0.3 (or is it really 4.0.2?) things have been running smoothly.

**Comment by ralphcallaway** [comment]
@dbleier yep, it's pretty stable, that's just me being overly cautious since i thought the last release was stable before this bug got reported, there are only two changes

and regarding version, v4.0.3 we just forgot to update the version text, it's not really changeable once released, so forever more v4.0.3 will show as v4.0.2 in the about page
